### PR TITLE
fix: Change lower case from bash variable expansion to tr

### PR DESCRIPTION
**Pull request by lazize**
### Context 

Bash variable expansion doesn't work on old bash version from macOS

### Description

Change lower case transformation from bash variable expansion to `tr '[:upper:]' '[:lower:]'` command.
Fix issue #1058 

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by lazize** [comment]
extra715
extra780
extra785
extra7101
check_sample

File: checks/check_extra7101 [hunk 1]
```diff
@@ -25,27 +25,27 @@ CHECK_DOC_extra7101='https://docs.aws.amazon.com/elasticsearch-service/latest/de
...
+    if [[ "${LIST_OF_DOMAINS}" ]]; then
+      for domain in ${LIST_OF_DOMAINS}; do
+        AUDIT_LOGS_ENABLED=$("${AWSCLI}" es describe-elasticsearch-domain-config --domain-name "${domain}" ${PROFILE_OPT} --region "${regx}" --query 'DomainConfig.LogPublishingOptions.Options.AUDIT_LOGS.Enabled' --output text 2>&1)
+        if [[ $(echo "${AUDIT_LOGS_ENABLED}" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+            textInfo "${regx}: Access Denied trying to get ES domain config for ${domain}" "${regx}"
             continue
         fi
-        if [[ "${AUDIT_LOGS_ENABLED,,}" == "true" ]];then
-          textPass "$regx: Amazon ES domain $domain AUDIT_LOGS enabled" "$regx" "$domain"
+        if [[ $(echo "${AUDIT_LOGS_ENABLED}" | tr '[:upper:]' '[:lower:]') == "true" ]]; then
```
Comment (n4ch04) [inline checks/check_extra7101]: Could you change this to 
`tr '[:upper:]' '[:lower:]' <<< "${AUDIT_LOGS_ENABLED}"`

Comment (n4ch04) [inline checks/check_extra7101]: Could you change this to 
`tr '[:upper:]' '[:lower:]' <<< "${AUDIT_LOGS_ENABLED}"`

File: checks/check_extra715 [hunk 2]
```diff
@@ -24,34 +24,35 @@ CHECK_DOC_extra715='https://docs.aws.amazon.com/elasticsearch-service/latest/dev
...
+        SLOWLOG_ENABLED=$("${AWSCLI}" es describe-elasticsearch-domain-config --domain-name "${domain}" ${PROFILE_OPT} --region "${regx}" --query 'DomainConfig.LogPublishingOptions.Options.[SEARCH_SLOW_LOGS.Enabled, INDEX_SLOW_LOGS.Enabled]' --output text 2>&1)
+        if [[ $(echo "${SLOWLOG_ENABLED}" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+            textInfo "${regx}: Access Denied trying to get ES domain config for ${domain}" "${regx}"
             continue
         fi
-        read SEARCH_SLOWLOG_ENABLED INDEX_SLOWLOG_ENABLED <<< "$SLOWLOG_ENABLED" && {
-          if [[ "${SEARCH_SLOWLOG_ENABLED,,}" == "true" ]];then
-            textPass "$regx: Amazon ES domain $domain SEARCH_SLOW_LOGS enabled" "$regx" "$domain"
+        read -r SEARCH_SLOWLOG_ENABLED INDEX_SLOWLOG_ENABLED <<< "$SLOWLOG_ENABLED" && {
+          if [[ $(echo "${SEARCH_SLOWLOG_ENABLED}" | tr '[:upper:]' '[:lower:]') == "true" ]]; then
```
Comment (n4ch04) [inline checks/check_extra715]: Same as in previous
`tr '[:upper:]' '[:lower:]' <<< "${SEARCH_SLOWLOG_ENABLED}"`

Comment (n4ch04) [inline checks/check_extra715]: Same as in previous
`tr '[:upper:]' '[:lower:]' <<< "${SEARCH_SLOWLOG_ENABLED}"`

File: checks/check_extra715 [hunk 3]
```diff
@@ -24,34 +24,35 @@ CHECK_DOC_extra715='https://docs.aws.amazon.com/elasticsearch-service/latest/dev
...
+            textPass "${regx}: Amazon ES domain ${domain} SEARCH_SLOW_LOGS enabled" "${regx}" "${domain}"
           else
-            textFail "$regx: Amazon ES domain $domain SEARCH_SLOW_LOGS disabled!" "$regx" "$domain"
+            textFail "${regx}: Amazon ES domain ${domain} SEARCH_SLOW_LOGS disabled!" "${regx}" "${domain}"
           fi
-          if [[ "${INDEX_SLOWLOG_ENABLED,,}" == "true" ]];then
-            textPass "$regx: Amazon ES domain $domain INDEX_SLOW_LOGS enabled" "$regx" "$domain"
+
+          if [[ $(echo "${INDEX_SLOWLOG_ENABLED}" | tr '[:upper:]' '[:lower:]') == "true" ]]; then
```
Comment (n4ch04) [inline checks/check_extra715]: Same as in previous 
`tr '[:upper:]' '[:lower:]' <<< "${INDEX_SLOWLOG_ENABLED}"`

Comment (n4ch04) [inline checks/check_extra715]: Same as in previous 
`tr '[:upper:]' '[:lower:]' <<< "${INDEX_SLOWLOG_ENABLED}"`

File: checks/check_extra780 [hunk 4]
```diff
@@ -24,27 +24,27 @@ CHECK_DOC_extra780='https://docs.aws.amazon.com/elasticsearch-service/latest/dev
...
+    if [[ "${LIST_OF_DOMAINS}" ]]; then
+      for domain in ${LIST_OF_DOMAINS}; do
+        CHECK_IF_COGNITO_ENABLED=$("${AWSCLI}" es describe-elasticsearch-domain --domain-name "${domain}" ${PROFILE_OPT} --region "${regx}" --query 'DomainStatus.CognitoOptions.Enabled' --output text 2>&1)
+        if [[ $(echo "${CHECK_IF_COGNITO_ENABLED}" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+            textInfo "${regx}: Access Denied trying to get ES domain ${domain}" "${regx}"
             continue
         fi
-        if [[ "${CHECK_IF_COGNITO_ENABLED,,}" == "true" ]];then
-          textPass "$regx: Amazon ES domain $domain has Amazon Cognito authentication for Kibana enabled" "$regx" "$domain"
+        if [[ $(echo "${CHECK_IF_COGNITO_ENABLED}" | tr '[:upper:]' '[:lower:]') == "true" ]]; then
```
Comment (n4ch04) [inline checks/check_extra780]: Same as in previous
`tr '[:upper:]' '[:lower:]' <<< "${CHECK_IF_COGNITO_ENABLED}"`

Comment (n4ch04) [inline checks/check_extra780]: Same as in previous
tr '[:upper:]' '[:lower:]' <<< "${CHECK_IF_COGNITO_ENABLED}"

File: checks/check_extra785 [hunk 5]
```diff
@@ -29,28 +29,28 @@ CHECK_CAF_EPIC_extra785='Infrastructure Security'
...
+        CHECK_IF_UPDATE_AVAILABLE_AND_VERSION=$("${AWSCLI}" es describe-elasticsearch-domain --domain-name "${domain}" ${PROFILE_OPT} --region "${regx}" --query 'DomainStatus.[ServiceSoftwareOptions.UpdateAvailable,ElasticsearchVersion]' --output text 2>&1)
+        if [[ $(echo "${CHECK_IF_UPDATE_AVAILABLE_AND_VERSION}" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+            textInfo "${regx}: Access Denied trying to get ES domain ${domain}" "${regx}"
             continue
         fi
-        read update_status es_version <<< "$CHECK_IF_UPDATE_AVAILABLE_AND_VERSION" &&
-        if [[ ${update_status,,} != "false" ]];then
-          textInfo "$regx: Amazon ES domain $domain v$es_version has updates available" "$regx" "$domain"
+        read -r update_status es_version <<< "${CHECK_IF_UPDATE_AVAILABLE_AND_VERSION}" &&
+        if [[ $(echo "${update_status}" | tr '[:upper:]' '[:lower:]') != "false" ]]; then
```
Comment (n4ch04) [inline checks/check_extra785]: Same as in previous: 
`tr '[:upper:]' '[:lower:]' <<< "${CHECK_IF_COGNITO_ENABLED}"`

Comment (n4ch04) [inline checks/check_extra785]: Same as in previous: 
`tr '[:upper:]' '[:lower:]' <<< "${CHECK_IF_COGNITO_ENABLED}"`

**Review by n4ch04 — CHANGES_REQUESTED** [review]

**Comment by n4ch04** [comment]
@lazize, I have requested a few changes more related with the style we are using in the project, please check it out and thank you for your contribution !!

**Review by n4ch04 — APPROVED** [review]

**Comment by n4ch04** [comment]
Thanks for all !! @lazize

**Pull request merged**